17HW/2_task/src/tests/2_task.test.ts

```diff
@@ -0,0 +1,197 @@
+    const errorMessageLoginSelector = '#errorMessage';
+    const successMessageLoginSelector = '#successMessage';
+    const loginSelectors: ICredentials = {
+        login: '#userName',
+        password: '#password'
+    };
+    const headerRegSelector = '#registerForm';
+    const headerLoginSelector = '#loginForm';
+
+    //credentials
+    const correctCredentials: ICredentials[] = [
+        {   login: 'AB1',
+            password: 'AbCd1234'
+        },
+        {   login: 'tqgkits!l lnktmwasqc',
+            password: 'rftxw#rV bobfeYtRngG'
+        },
+        {   login: 'Алена',
+            password: 'Мойсложныйпароль'
+        },
+        {   login: 'AB2',
+            password: ' AbCd1234'
+        },
+        {   login: 'AB3',
+            password: 'AbCd1234 '
+        },
+        {   login: 'AGSBWBHECHHCHHEHEHCHCHEHCHEHEHHC1hhdhhqw',
+            password: 'AbCd1234 '
+        }
+    ];
+    const incorrectCredentials: ICredentials[] = [
+        {   login: 'AB4',
+            password: '        '//пробелы
+        },
+        {   login: 'Login',
+            password: 'ABCDEFGHIJKLM'//только заглавные
+        },
+        {   login: 'LO', //короткий
+            password: 'AbCGh26j1'
+        },
+        {   login: '   ', //пробелы
+            password: 'AbCd1234'
+        },
+        {   login: ' AB5', //пробелы
+            password: 'AbCd1234'
+        },
+        {   login: 'AB6 ', //пробелы
+            password: 'AbCd1234'
+        },
+        {   login: 'AB7',
+            password: 'FBu123i' //короткий
+        },
+        {   login: 'AGSBWBHECHHCHHEHEHCHCHEHCHEHEHHC1hhdhhqwdfsdfs', //длинный
+            password: 'AbC76YjIIo'
+        },
+        {   login: 'AB8',
+            password: 'NLsyDrnRuvEcLIbilcqJo' //длинный
+        },
+    ];
+
+
+    //before and after tests
+    before(async function () {
+        await browser.maximizeWindow();
+      });
+    
+    beforeEach(async function () {
+        await browser.url(url);
+    });
+    afterEach(() => {
+        browser.execute('window.localStorage.clear()');
+    });
+
+
+    //tests
+    it('NewUser', async function () {
+        await $(goToRegisterButtonSelector).click();
+        for (let item of correctCredentials) {
+            await $(registrSelectors.login).setValue(item.login);
+            await $(registrSelectors.password).setValue(item.password);
+            await $(sumbitRegister).click();
+            const actualText = await $(errorMessageRegisterSelector).getText();
+            expect(actualText).toContain(successNotification);
+            const stringFromStorage = await browser.execute((login) => {
+                return localStorage.getItem(login);
+            }, item.login);
+            const objectFromStorage = JSON.parse(stringFromStorage as string);
+            expect(objectFromStorage.name).toBe(item.login);
+            expect(objectFromStorage.password).toBe(item.password);
+        }
+    });
```

Comment:
Из названия теста мьі должньі понимать, что происходит, тут закрадьівается вопрос - "что `NewUser`" ?
Как вариант, если мьі пьітаемся воспользоваться параметризацией
```suggestion
  for (const { login, password } of correctCredentials) {
    it('Should register user with login: ${login} and password: ${password}', async function () {
        await $(goToRegisterButtonSelector).click();
            await $(registrSelectors.login).setValue(item.login);
            await $(registrSelectors.password).setValue(item.password);
            await $(sumbitRegister).click();
            const actualText = await $(errorMessageRegisterSelector).getText();
            expect(actualText).toContain(successNotification);
            const stringFromStorage = await browser.execute((login) => {
                return localStorage.getItem(login);
            }, item.login);
            const objectFromStorage = JSON.parse(stringFromStorage as string);
            expect(objectFromStorage.name).toBe(item.login);
            expect(objectFromStorage.password).toBe(item.password);
         });
        }
```